### PR TITLE
Change allow mutation wording

### LINKED ISSUE #6181: Allow Mutation - Request to re-add
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

Hi all - first of all I'm very excited for the release of Pydantic V2! The features removed in general I think look like very good decisions. There was only one feature on the list that I felt I would miss - `allow_mutation`, in Config. I've used this in some projects before to allow users to modify large already generated models, then reinsert the modified version, rather than have to regenerate the model anew, which may have many attributes. I'd be interested to hear the reasons for its removal, or if there are any alternatives for this use case, that wouldn't involve reassigning all attributes of the model.

Here's a demonstration of the use case in V1 syntax:
```py
from pydantic import BaseModel
class NewBase(BaseModel):
    class Config:
         allow_mutation = True

class Submodel1(NewBase):
    attr1: str = 'test' # Imagine many more attributes here

class Submodel2(NewBase):
    attr2: str = 'test' # Imagine many more attributes here

class Wrapper(NewBase):
    sub1: Submodel1
    sub2: Submodel2

inst = Wrapper(sub1=Submodel1(), sub2=Submodel2())
f(inst) # Some Function
inst.sub1.attr1 = 'new'
f(inst) # Some Function
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [X] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

### PR DESCRIPTION
**Pull request by mark-todd**
## Change Summary

Changed wording of blog post to show frozen can be used instead of allow_mutation.

## Related issue number

fix #6181

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: changes/6181-mark-todd.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Updated docs to show frozen can be used instead of allow_mutation
```
Comment (hramezani) [inline changes/6181-mark-todd.md:1]: We don't need this file for changes like this
Please remove it

Comment (mark-todd) [reply changes/6181-mark-todd.md]: File removed

File: docs/blog/pydantic-v2-alpha.md [hunk 2]
```diff
@@ -133,7 +133,7 @@ a full migration guide, but for now the following pointers should be some help w
 
 The following config settings have been removed:
 
-* `allow_mutation`.
+* `allow_mutation` — this has been removed. You should be able to use [frozen](https://docs.pydantic.dev/dev-v2/api/config/#pydantic.config.ConfigDict) equivalently (inverse of current use).
```
Comment (hramezani) [inline docs/blog/pydantic-v2-alpha.md:136]: We don't need this change in the blog post.

Please revert this change and change the migration guide in https://github.com/pydantic/pydantic/blob/c2fe0b87e1cbd09a58fb30be5ec6bbb372280c45/docs/migration.md?plain=1#L145

Comment (hramezani) [reply docs/blog/pydantic-v2-alpha.md:136]: Also, it's better to use a relative link
[frozen](api/config.md#pydantic.config.ConfigDict)

Comment (mark-todd) [reply docs/blog/pydantic-v2-alpha.md:136]: Sure - would it not be helpful in the blog post as well? Appreciate the migration guide is the more important of the two

Comment (hramezani) [reply docs/blog/pydantic-v2-alpha.md:136]: Ok, let's do it in both places

**Comment by mark-todd** [comment]
working on link fix

File: docs/blog/pydantic-v2-alpha.md (hunk 2 above)
Comment (mark-todd) [reply docs/blog/pydantic-v2-alpha.md]: Resolved

**Review by hramezani — APPROVED** [review]
